packages/core/src/modules/connections/services/helpers.ts

```diff
@@ -82,8 +82,14 @@ export function convertToNewDidDocument(didDoc: DidDoc): DidDocument {
 }
 
 function normalizeId(fullId: string): `#${string}` {
-  const [, id] = fullId.split('#')
+  // Some old dids use `;` as the delimiter for the id. If we can't find a `#`
+  // and a `;` exists, we will parse everything after `;` as the id.
+  if (!fullId.includes('#') && fullId.includes(';')) {
```

Comment:
This parser only works if there is a single `;` in the id. Is that correct?

Reply:
The only usage I've seen is with one `;`. I think it was used before the `#` and you can only include one. What we could do is leave the other ones in place (like url parsers do):

![image](https://user-images.githubusercontent.com/23165168/174558626-559323f5-858c-4598-8f97-96bb5fcaf9db.png)

So we would get `#indy;anotherword` instead of `#indy` if the id was `something;indy;anotherword`. Do you think that is better?

Reply:
I think leaving them behind, which follows the url parser you showed, is good. I am very unaware of the security implications (likely none) here though.